docs/source/features/memory_optimizations.rst

```diff
@@ -5,6 +5,80 @@ Parallelism
 -----------
 Refer to :doc:`Parallelism <./parallelisms>`.
 
+
+Mixture of Experts
+------------------
+
+Overview
+^^^^^^^^
+
+NeMo supports Mixture of Experts (MoE) in the transformer layer for NLP models.
```

Comment:
It's not limited to NLP. Maybe we can just say "... in the feedforward block of a transformer layer."